scripts/python/assets-automation/generate_assets_report.py

```diff
@@ -0,0 +1,306 @@
+    repo = get_repo(language)
+    print(f"Evaluating repo for {language} @ {repo}", end="...")
+
+    try:
+        target_folder = os.path.join(repo, "sdk", "**", "package.json")
+        result = ScanResult(language)
+
+        results = glob.glob(target_folder, recursive=True)
+
+        result.packages = [
+            os.path.basename(os.path.dirname(pkg))
+            for pkg in results
+            if "samples" not in os.path.normpath(pkg).split(os.sep)
+            and os.path.basename(os.path.dirname(pkg)) != "samples-react"
+        ]
+
+        for pkg in results:
+            evaluation = evaluate_js_package(pkg)
+            if evaluation == 1:
+                result.packages_using_proxy.append(os.path.basename(os.path.dirname(pkg)))
+            elif evaluation == 2:
+                result.packages_using_proxy.append(os.path.basename(os.path.dirname(pkg)))
+                result.packages_using_external.append(os.path.basename(os.path.dirname(pkg)))
+        print(YES)
+    except Exception as e:
+        print(NO)
+
+    return result
+
+
+def generate_detailed_table(origin: ScanResult, package_set: List[str]):
+    result = TABLE_HEADER
+    for package in package_set:
+        transitioned = YES if package in origin.packages_using_proxy else NO
+        externalized = YES if package in origin.packages_using_external else NO
+
+        table_row = TABLE_LAYER.format(package.replace("\\", "/"), transitioned, externalized)
+        result += table_row
+
+    return result
+
+
+def write_output(result: ScanResult) -> None:
+    with open(result.language.lower() + ".md", "w", encoding="utf-8") as f:
+        date = datetime.date.today()
+        time_of_day = datetime.datetime.today().strftime("%I:%M%p")
+        f.writelines(
+            f"# {result.language} Transition Details as of {date}@{time_of_day} {datetime.datetime.today().astimezone().tzname()}"
+        )
+
+        if result.packages:
+            # batch by sets of 20
+            batch_size = (len(result.packages) // 2) + (len(result.packages) % 2)
+
+            table_set_1 = result.packages[0:batch_size]
+            table_set_2 = result.packages[batch_size:]
+
+            document_addition = DOCUMENT.format(
+                generate_detailed_table(result, table_set_1), generate_detailed_table(result, table_set_2)
+            )
+
+            f.write(document_addition)
+
+
+def write_summary(results: List[ScanResult]) -> None:
+    with open("summary.md", "w", encoding="utf-8") as f:
+        f.writelines(f"# Test Proxy Transition Summary - {datetime.date.today()}")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description="""
+      Generates a markdown report that summarizes the the status of the transition to the test-proxy and externalized assets.
+      """
+    )
+    parser.parse_args()
+
+    # python = generate_python_report()
+    # write_output(python)
+
+    # js = generate_js_report()
+    # write_output(js)
```

Comment:
Once one of these is completed I comment it to increase dev-cycle speed. Uncomment when finished.